### PR TITLE
Add namespace rate / count limiter interceptor

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add namespace rate limiter RPC interceptor
* Add namespace count limiter RPC interceptor
* Refactor frontend handler

**Why?**
N/A

**How did you test it?**
N/A

**Potential risks**
N/A

### CODE REVIEW

File: service/history/transferQueueActiveTaskExecutor.go [hunk 1]
```diff
@@ -1152,13 +1152,6 @@ func (t *transferQueueActiveTaskExecutor) requestCancelExternalExecutionWithRetr
 	}
 
 	err := backoff.Retry(op, persistenceOperationRetryPolicy, common.IsPersistenceTransientError)
-
-	if _, ok := err.(*serviceerror.CancellationAlreadyRequested); ok {
-		// err is CancellationAlreadyRequested
-		// this could happen if target workflow cancellation is already requested
-		// mark as success
-		return nil
-	}
```
Comment (wxing1292) [inline service/history/transferQueueActiveTaskExecutor.go:1161]: removing dead code

File: common/rpc/interceptor/namespace_count_limit.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	"context"
+	"sync"
+	"sync/atomic"
+
+	"go.temporal.io/api/serviceerror"
+	"google.golang.org/grpc"
+)
+
+var (
+	NamespaceCountLimitServerBusy = serviceerror.NewResourceExhausted("namespace count limit exceeded")
```
Comment (alexshtin) [inline common/rpc/interceptor/namespace_count_limit.go:37]:
```suggestion
	ErrNamespaceCountLimitServerBusy = serviceerror.NewResourceExhausted("namespace count limit exceeded")
```

File: common/rpc/interceptor/namespace_count_limit.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+) *NamespaceCountLimitInterceptor {
+	return &NamespaceCountLimitInterceptor{
+		countFn: countFn,
+		tokens:  tokens,
+
+		namespaceToCount: make(map[string]*int32),
+	}
+}
+
+func (i *NamespaceCountLimitInterceptor) Intercept(
```
Comment (alexshtin) [inline common/rpc/interceptor/namespace_count_limit.go:64]:
```suggestion
func (ni *NamespaceCountLimitInterceptor) Intercept(
```

File: common/rpc/interceptor/namespace_count_limit.go (hunk 2 above)
Comment (alexshtin) [reply common/rpc/interceptor/namespace_count_limit.go:37]: and few other places.

**Review by alexshtin — APPROVED** [review]

**Pull request merged**